service/src/test/java/io/camunda/service/ProcessInstanceServiceTest.java

```diff
@@ -143,9 +145,10 @@ public void shouldReturnProcessInstanceSequenceFlows() {
   public void shouldReturnProcessInstanceByKey() {
     // given
     final var key = 123L;
-    final var entity = mock(ProcessInstanceEntity.class);
-    when(entity.processInstanceKey()).thenReturn(key);
-    when(entity.processDefinitionId()).thenReturn("processId");
+    final var entity =
+        Instancio.of(ProcessInstanceEntity.class)
+            .set(field(ProcessInstanceEntity::getProcessInstanceKey), key)
```

Comment:
@Vishalup29 I see where the issue is, you are trying to call the getter here, but the `ProcessInstanceEntity` is a record and they provide different accessor methods.

Can you adjust all the places to use the proper accessors? Can you also try to compile the codebase locally on your machine? (You find the howto in our contribution [guide](https://github.com/camunda/camunda/blob/main/CONTRIBUTING.md)) 💪 


```suggestion
            .set(field(ProcessInstanceEntity::processInstanceKey), key)
```

Reply:
Thanks @eppdot , fixed it, for some reason it couldn't compile locally on my machine due to some jdk issue. This time i was able to fix it and compiled locally, it was successful.